src/features/cosoul/api/cosoul.ts

```diff
@@ -65,6 +89,13 @@ export const setOnChainPGIVE = async (tokenId: number, amt: number) => {
   return await contract.setSlot(PGIVE_SLOT, amount, tokenId, gasSettings);
 };
 
```

Comment:
create simple method `export const setBatchOnChainPGIVE = async ([tokenId: number, amt: number]) => {`

 that takes same type as setOnChainPGive but array of it, and all details are hidden inside.